validation/validation_data.py

```diff
@@ -0,0 +1,345 @@
+    lat_avg = sum([node.p.lat for node in NODES.values()]) / len(NODES)
+    lon_avg = sum([node.p.lon for node in NODES.values()]) / len(NODES)
+    map_center = [lat_avg, lon_avg]
+    
+    mymap = folium.Map(location=map_center, zoom_start=17)
+
+    for edge in EDGES:
+        folium.PolyLine(
+            locations=[(p.lat, p.lon) for p in edge.get_line()],
+            color=get_line_color([segment.pci for segment in edge.segments]),
+            weight=3,
+            opacity=0.6,
+            popup=[segment.get_text() for segment in edge.segments]
+        ).add_to(mymap)
+    
+    create_dir(MAPS_DIR)
+    mymap.save(QUALITY_MAP_FILE)
+    return lat_avg, lon_avg
+
+
+def get_pci_color(pci: float):
+    if pci == 0:
+        return 'black'
+    if pci < 20:
+        return 'red'
+    if pci < 40:
+        return 'orange'
+    if pci < 60:
+        return 'yellow'
+    if pci < 80:
+        return 'green'
+    return 'blue'
+
+
+def get_line_color(pcis: list[float]):
+    if not pcis:
+        return 'grey'
+    if len(pcis) == 1:
+        return get_pci_color(pcis[0])
+    return 'pink'
+
+
+def normalize_street_name(name: str):
+    if not name:
+        return name
+    if name in MAPPED_NAMES:
+        return MAPPED_NAMES[name]
+    name = re.sub(r'\bAvenue\b', 'AVE', name)
+    name = re.sub(r'\bStreet\b', 'ST', name)
+    name = re.sub(r'\bRoad\b', 'RD', name)
+    name = re.sub(r'\bDrive\b', 'DR', name)
+    name = re.sub(r'\bDriveway\b', 'DRWY', name)
+    name = re.sub(r'\bPlace\b', 'PL', name)
+    name = re.sub(r'\bLane\b', 'LN', name)
+    name = re.sub(r'\bCourt\b', 'CT', name)
+    name = re.sub(r'\bWay\b', 'WAY', name)
+    name = re.sub(r'\bTerrace\b', 'TR', name)
+    name = re.sub(r'\bEast\b', 'E', name)
+    name = re.sub(r'\bWest\b', 'W', name)
+    name = re.sub(r'\bSouth\b', 'S', name)
+    name = re.sub(r'\bNorth\b', 'N', name)
+    name = re.sub(r'\bSoutheast\b', 'SE', name)
+    name = re.sub(r'\bSouthwest\b', 'SW', name)
+    name = re.sub(r'\bNortheast\b', 'NE', name)
+    name = re.sub(r'\bNorthwest\b', 'NW', name)
+    name = re.sub(r'\bBoulevard\b', 'BLVD', name)
+    name = re.sub(r'\bShip Canal Bridge\b', 'SHIP CANAL TRL', name)
+    return name.upper()
+
+
+def create_dir(dir):
+    if not os.path.exists(dir):
+        os.makedirs(dir)
+
+
+NODES: dict[int, Node] = {}
+EDGES: list[Edge] = []
+SAMPLE_EDGES: dict[tuple[str, str, str], Edge] = {}
+STREET_SEGMENTS: dict[str, StreetSegment] = {}
+STREET_NAMES = set()
+
+MAPPED_NAMES = {
+    'East Terrace Street': 'E TERRACE ST',
+    'Sound View Terrace West': 'SOUND VIEW TER W',
+    'South Lane Street': 'S LANE ST',
+    'West Montlake Place East': 'WEST MONTLAKE PL E',
+    'East Montlake Place East': 'EAST MONTLAKE PL E',
+    'East North Street': 'E NORTH ST',
+    'West Laurelhurst Drive Northeast': 'WEST LAURELHURST DR NE',
+    'East Laurelhurst Drive Northeast': 'EAST LAURELHURST DR NE',
+    'Terrace Street': 'TERRACE ST',
+}
+
+SKIPPED_ROADS = [
```

Comment:
why are these skipped?

Reply:
These roads exist in OSM but not in Seattle road dataset or I was not able to do the mapping. So I had them skipped.

Reply:
add a comment explaining this